src/main/java/edu/ucla/library/prl/harvester/Job.java

```diff
@@ -0,0 +1,178 @@
+
+    /**
+     * The JSON key for the institution ID.
+     */
+    static final String INSTITUTION_ID = "institutionId";
+
+    /**
+     * The JSON key for the repository base URL.
+     */
+    static final String REPOSITORY_BASE_URL = "repositoryBaseUrl";
+
+    /**
+     * JSON key for the metadata prefix.
+     */
+    static final String METADATA_PREFIX = "metadataPrefix";
+
+    /**
+     * JSON key for the list of sets.
+     */
+    static final String SETS = "sets";
+
+    /**
+     * JSON key for the schedule.
+     */
+    static final String SCHEDULE_CRON_EXPRESSION = "scheduleCronExpression";
+
+    /**
+     * JSON key for the last successful run.
+     */
+    static final String LAST_SUCCESSFUL_RUN = "lastSuccessfulRun";
+
+    /**
+     * The identifier of the institution that this job should be associated with.
+     */
+    private final int myInstitutionID;
+
+    /**
+     * The base URL of the OAI-PMH repository.
+     */
+    private final URL myRepositoryBaseURL;
+
+    /**
+     * The list of sets to harvest.
+     */
+    private final Optional<List<String>> mySets;
+
+    /**
+     * The schedule on which this job should be run.
+     */
+    private final CronExpression myScheduleCronExpression;
+
+    /**
+     * The timestamp of the last successful run of this job; will be empty at first.
+     */
+    private final Optional<ZonedDateTime> myLastSuccessfulRun;
+
+    /**
+     * Instantiates a job.
+     *
+     * @param anInstitutionID The identifier of the institution that this job should be associated with
+     * @param aRepositoryBaseURL The base URL of the OAI-PMH repository
+     * @param aSets The list of sets to harvest; if empty, assume all sets should be harvested
+     * @param aScheduleCronExpression The schedule on which this job should be run
+     * @param aLastSuccessfulRun The timestamp of the last successful run of this job; will be null at first
+     */
+    public Job(final int anInstitutionID, final URL aRepositoryBaseURL, final List<String> aSets,
+            final CronExpression aScheduleCronExpression, final ZonedDateTime aLastSuccessfulRun) {
+        myInstitutionID = anInstitutionID;
+        myRepositoryBaseURL = Objects.requireNonNull(aRepositoryBaseURL);
+        mySets = Optional.ofNullable(aSets);
+        myScheduleCronExpression = Objects.requireNonNull(aScheduleCronExpression);
+        myLastSuccessfulRun = Optional.ofNullable(aLastSuccessfulRun);
+    }
+
+    /**
+     * Instantiates a job from its JSON representation.
+     * <p>
+     * <b>This constructor is meant to be used only by generated service proxy code!</b>
+     * {@link #Job(int, URL, List, CronExpression, ZonedDateTime)} should be used everywhere else.
+     *
+     * @param aJsonObject A job represented as JSON
+     * @throws InvalidJobJsonException If the JSON representation is invalid
+     */
+    @SuppressWarnings({ "PMD.AvoidCatchingNPE", "PMD.AvoidCatchingGenericException" })
```

Comment:
If we disagree in principle with these PMD warnings, we could disable them for the project. I tend to think of SuppressWarnings as a way to let technical debt exist until it can be cleaned up. This would be a new ticket.

Reply:
I know yesterday you mentioned using the requireNonNull to throw NPEs instead of doing the below for every field:

```
final String name = aJsonObject.getString(NAME);

if (name == null) {
    throw new InvalidInstitutionException("Missing 'name' property");
} else {
    myName = name;
}
```
That could also be put into a method though:

```
checkProperty(aJsonObject.getString(NAME));

public String checkProperty(final String aName) {
  if (aName == null) {
    throw new InvalidInstitutionException(aName);
  } else {
    return aName;
  }
}
```
fwiw...

Reply:
Okay, fair enough -- I shouldn't make an _exception_ for myself